controllers/infrastructure/byoadmission_controller_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/test/utils/csr"
+	certv1 "k8s.io/api/certificates/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("Controllers/ByoadmissionController", func() {
+	var (
+		ctx context.Context
+		err error
+		CSR *certv1.CertificateSigningRequest
+	)
+
+	It("should return error for non-existent CSR", func() {
+		// Start recoincilation for a non-existing CSR
+		objectKey := types.NamespacedName{Name: defaultByoHostName}
+		_, err = byoAdmissionReconciler.Reconcile(ctx, reconcile.Request{NamespacedName: objectKey})
+		Expect(err).To(BeNil())
+	})
+
+	Context("When a CSR request is made", func() {
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			// Create a CSR resource for each test
+			CSR, err = csr.CreateCSRResource(defaultByoHostName, "byoh:hosts")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should approve the Byoh CSRs", func() {
+			// Create a dummy CSR request
+			_, err = clientSetFake.CertificatesV1().CertificateSigningRequests().Create(ctx, CSR, v1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			// Start recoincilation
+			objectKey := types.NamespacedName{Name: defaultByoHostName}
+			_, err = byoAdmissionReconciler.Reconcile(ctx, reconcile.Request{NamespacedName: objectKey})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Fetch the updated CSR
+			var updateByohCSR *certv1.CertificateSigningRequest
+			updateByohCSR, err = clientSetFake.CertificatesV1().CertificateSigningRequests().Get(ctx, defaultByoHostName, v1.GetOptions{})
```

Comment:
```suggestion
			updateByohCSR, err := clientSetFake.CertificatesV1().CertificateSigningRequests().Get(ctx, defaultByoHostName, v1.GetOptions{})
```

Reply:
`updateByohCSR` is declared with var instead of declaration operator  (:=) as `err` is globally defined for all tests and using declaration operator causes linting errors.